strax/storage/common.py

```diff
@@ -551,23 +557,46 @@ def save_from(self, source: typing.Generator, rechunk=True, executor=None):
         pending = []
         exhausted = False
         chunk_i = 0
+
+        run_id = self.md['run_id']
+        _is_super_run = run_id.startswith('_')
         try:
+            last_chunk_end = None
             while not exhausted:
                 chunk = None
+
                 try:
                     if rechunk and self.allow_rechunk:
                         while (chunk is None or
                                 chunk.data.nbytes < chunk.target_size_mb*1e6):
+                            next_chunk = next(source)
+
+                            if _is_super_run:
+                                # If we are creating a superrun, we load data from subruns
+                                # and the loaded subrun chunk becomes a superun chunk:
+                                next_chunk = strax.transform_chunk_to_superrun_chunk(run_id, 
+                                                                                     next_chunk, 
+                                                                                     last_chunk_end)
+                                last_chunk_end = next_chunk.end
+                            
```

Comment:
I might now see why you made this transform_chunk_to_superrun_chunk function (see above). But would the issue you are trying to solve not much easier be addressed in `continuity_check`?

Reply:
Lets keep discussing in the thread above.